r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -0,0 +1,172 @@
+    # so we can't use expect_dplyr_equal() for the tests below
+    expect_equal(
+      df %>%
+        Table$create() %>%
+        transmute(x = sub("O", "u", x, ignore.case = TRUE, fixed = fixed)) %>%
+        collect(),
+      tibble(x = c("Fuo", "bar"))
+    )
+    expect_equal(
+      df %>%
+        Table$create() %>%
+        transmute(x = gsub("o", "u", x, ignore.case = TRUE, fixed = fixed)) %>%
+        collect(),
+      tibble(x = c("Fuu", "bar"))
+    )
+    expect_equal(
+      df %>%
+        Table$create() %>%
+        transmute(x = sub("^B.+", "baz", x, ignore.case = TRUE, fixed = fixed)) %>%
+        collect(),
+      if (fixed) tibble(x = c("Foo", "bar")) else tibble(x = c("Foo", "baz"))
+    )
+
+  }
+
+})
+
+test_that("str_replace and str_replace_all", {
+  df <- tibble(x = c("Foo", "bar"))
+
+  library(stringr)
+
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_replace_all(x, regex("^F"), "baz")) %>%
+      collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_replace(x, regex("^f[A-Z]{2}", ignore_case = TRUE), "baz")) %>%
+      collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_replace_all(x, fixed("o"), "u")) %>%
+      collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_replace(x, fixed("O"), "u")) %>%
+      collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_replace(x, fixed("O", ignore_case = TRUE), "u")) %>%
+      collect(),
+    df
+  )
+
+})
+
+test_that("backreferences", {
+  df <- tibble(x = c("Foo", "bar"))
+
+  expect_dplyr_equal(
+    input %>%
+      transmute(desc = sub(
+        "(?:https?|ftp)://([^/\r\n]+)(/[^\r\n]*)?",
+        "path `\\2` on server `\\1`",
+        url
+        )
+      ) %>%
+      collect(),
+    tibble(url = "https://arrow.apache.org/docs/r/")
+  )
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = str_replace(x, regex("^(\\w)o(.*)", ignore_case = TRUE), "\\1\\2p")) %>%
+      collect(),
+    df
+  )
+})
+
+test_that("edge cases", {
+
+  # in case-insensitive fixed replace, test that "\\E" in the search string and
+  # backslashes in the replacement string are interpreted literally
+  expect_equal(
+    tibble(x = c("\\Q\\e\\D")) %>%
+      Table$create() %>%
+      transmute(x = sub("\\E", "\\L", x, ignore.case = TRUE, fixed = TRUE)) %>%
+      collect(),
+    tibble(x = c("\\Q\\L\\D"))
+  )
+
+  # test that a user's "(?i)" prefix does not break the "(?i)" prefix that's
+  # added in case-insensitive regex replace
+  expect_equal(
+    tibble(x = c("ABC")) %>%
+      Table$create() %>%
+      transmute(x = sub("(?i)^[abc]{3}$", "123", x, ignore.case = TRUE, fixed = FALSE)) %>%
+      collect(),
+    tibble(x = c("123"))
+  )
+
+})
+
+test_that("errors and warnings", {
+  skip_if_not_available("dataset")
+
+  df <- tibble(x = c("Foo", "bar"))
+  ds <- InMemoryDataset$create(df)
```

Comment:
Do these tests require a dataset? If so, can you comment why? If not, it's confusing that you're switching to a dataset here.

Reply:
I used a dataset to suppress the errors-to-warnings thing that `abandon_ship()` does with Tables and RecordBatches. I can just use a Table and use `expect_warning()` instead of `expect_error()`

Reply:
Changed in 7aee83c854ac87db17054a7692b35e267e81a0c0